src/GitHub.Api/Git/Repository.cs

```diff
@@ -309,273 +207,97 @@ private void CacheContainer_OnCacheInvalidated(CacheType cacheType)
+            });
         }
 
         private void RepositoryManagerOnGitLogUpdated(List<GitLogEntry> gitLogEntries)
         {
-            new ActionTask(TaskManager.Instance.Token, () => {
-                CurrentLog = gitLogEntries;
-            }) { Affinity = TaskAffinity.UI }.Start();
+            taskManager.RunInUI(() => cacheContainer.GitLogCache.Log = gitLogEntries);
         }
 
         private void RepositoryManagerOnGitLocksUpdated(List<GitLock> gitLocks)
         {
-            new ActionTask(TaskManager.Instance.Token, () => {
-                    CurrentLocks = gitLocks;
-                })
-                { Affinity = TaskAffinity.UI }.Start();
+            taskManager.RunInUI(() => cacheContainer.GitLocksCache.GitLocks = gitLocks);
         }
 
         private void RepositoryManagerOnRemoteBranchesUpdated(Dictionary<string, ConfigRemote> remotes,
             Dictionary<string, Dictionary<string, ConfigBranch>> branches)
         {
-            new ActionTask(TaskManager.Instance.Token, () => {
+            taskManager.RunInUI(() => {
                 cacheContainer.BranchCache.SetRemotes(remotes, branches);
-                Remotes = ConfigRemotes.Values.Select(GetGitRemote).ToArray();
-                RemoteBranches = RemoteConfigBranches.Values.SelectMany(x => x.Values).Select(GetRemoteGitBranch).ToArray();
-            }) { Affinity = TaskAffinity.UI }.Start();
+                cacheContainer.BranchCache.Remotes = ConfigRemotes;
+                cacheContainer.BranchCache.RemoteBranches = RemoteConfigBranches;
+            });
         }
 
         private void RepositoryManagerOnLocalBranchesUpdated(Dictionary<string, ConfigBranch> branches)
         {
-            new ActionTask(TaskManager.Instance.Token, () => {
+            taskManager.RunInUI(() => {
                 cacheContainer.BranchCache.SetLocals(branches);
-                UpdateLocalBranches();
-            }) { Affinity = TaskAffinity.UI }.Start();
-        }
-
-        private void UpdateLocks()
-        {
-            if (CurrentRemote.HasValue)
-            {
-                repositoryManager?.UpdateLocks();
-            }
-        }
-
-        private void UpdateLocalBranches()
-        {
-            LocalBranches = LocalConfigBranches.Values.Select(GetLocalGitBranch).ToArray();
-        }
-
-        private void ClearRepositoryInfo()
-        {
-            CloneUrl = null;
-            Name = null;
+                cacheContainer.BranchCache.LocalBranches = LocalConfigBranches;
+            });
         }
 
-        private GitBranch GetLocalGitBranch(ConfigBranch x)
+        private static GitBranch GetLocalGitBranch(string currentBranchName, ConfigBranch x)
         {
             var branchName = x.Name;
             var trackingName = x.IsTracking ? x.Remote.Value.Name + "/" + branchName : "[None]";
-            var isActive = branchName == CurrentBranchName;
-
+            var isActive = branchName == currentBranchName;
```

Comment:
BTW I did this because calling `CurrentBranchName` hits the cache and can trigger an invalidation request, which is not something we want to do when we're updating another cache.